spec/registry.md

```diff
@@ -1,11 +1,34 @@
 # MessageFormat 2.0 Default Function Registry
 
-This section describes the functions for which each implementation MUST provide
-a _function handler_ to be conformant with this specification.
-
-Implementations MAY implement additional _functions_ or additional _options_.
-In particular, implementations are encouraged to provide feedback on proposed
-_options_ and their values.
+This section defines the **standard** _functions_ which are REQUIRED for conformance with this specification,
+along with **optional** _functions_ that SHOULD be implemented to support
+additional functionality.
+
+Implementations MUST define and support each **standard** _function_ and
+MUST accept all _options_ and _option_ values defined as **standard** listed in this specification.
+
+When defined and supported by an implementation, each **optional** _function_ 
+MUST accept all _options_ and _option_ values listed as **standard** in this specification.
+
+Implementations SHOULD implement _options_ and _option_ values that are marked as **optional**.
+
+Implementations MAY implement _functions_ not defined in this specification.
+Such functions MUST use an implementation-defined _namespace_.
+In addition, implementations are encouraged to provide mechanisms for users to
+register and use user-defined _functions_ and their associated _functional handlers_.
+
+Implementations MAY implement additional _options_ not defined here.  
+Such _options_ MUST use an implementation-specific _namespace_.
+
+Implementations MAY implement additional _option_ values for _options_ defined here,
+although care needs to be exercised to ensure interoperability
+and to avoid collisions with future standardization.
+There is no namespace mechanism for _option_ values,
+however, the [stability policy](#stability-policy) for this specification 
+permits only the ranges a-z, A-Z, and 0-9 in _option_ values.
+Implementation-defined values SHOULD use a distinguishing character 
+or character sequence, such as by prefixing with a `_` U+005F LOW LINE,
+to ensure that they don't collide with future standardization.
```

Comment:
```suggestion
There is no namespace mechanism for _option_ values.
Implementation-defined _enum_ values SHOULD use a distinguishing character 
or character sequence, such as by prefixing with a `_` U+005F LOW LINE,
to ensure that they don't collide with future standardization.

For example, the option `compactDisplay` for `:currency` has option values `short` and `long`.
If an implementation adds a medium value, it should be distinguished, eg `_medium`.
```

Reply:
Why not recommend that these be `foo:` namespaced? As in, `compactDisplay=icu:medium`.

Reply:
I didn't think that we had namespaced option **values**.

Reply:
We don't have namespaces option values and the ABNF does not permit a colon in the unquoted value (there can be one in a literal, but the literal has to be quoted to have a colon).

Reply:
> Implementation-defined _enum_ values

`enum` isn't a defined term. I get that you're trying to avoid making people put underscores on values such as our digit option value or other normal data serializations.

Reply:
Maybe that could be omitted, or rephrased as something like:

Implementation-defined values that represent an enumeration of values such as {`short`, `medium`, `long`} SHOULD use a distinguishing character

Reply:
> We don't have namespaces option values and the ABNF does not permit a colon in the unquoted value (there can be one in a literal, but the literal has to be quoted to have a colon).

Oh, true. Then recommending a `_` prefix is probably a good idea.